vsphere/datadog_checks/vsphere/api.py

```diff
@@ -439,9 +440,12 @@ def get_vsan_metrics(self, cluster_nested_elts, entity_ref_ids, id_to_tags, star
                 )
             health_metrics.append(processed_health_metrics)
 
+            redapl_cluster_data = {'name': cluster_reference.name, 'vcenter_server': self.config.hostname}
             vsan_perf_query_spec = []
             for nested_id in nested_ids:
                 for entity_type in entity_ref_ids[id_to_tags[nested_id][0]]:
```

Comment:
```python
... entity_ref_ids[id_to_tags[nested_id][0]]:
```

 is this access completely safe?

Reply:
Yes, I create all of these dictionaries: `entity_ref_ids`,`id_to_tags`,`nested_id` in [this method](https://github.com/DataDog/integrations-core/blob/rahul/vsphere-to-evp/vsphere/datadog_checks/vsphere/vsphere.py#L590). I define that every collected resource be included in `id_to_tags` by their id, and I define that their "0" value be the type of the resource. Basically, I just want to see, for each resource within a given cluster, what type of resource it is (either a host or the cluster itself, for now), and then depending on the type of the resource, choose the entities (list of metrics) that are appropriate to find for it.